### PR TITLE
[TBDGen] Side-effects and asserts don't mix.

### PR DESCRIPTION
**Pull request by huonw**
Fixes rdar://problem/31343022.

### CODE REVIEW

**Comment by huonw** [comment]
@swift-ci Please smoke test and merge.

**Comment by gottesmm** [comment]
@huonw if your smoke test was a python failure, I think @erg fixed it. I am going to restart it for you.

**Comment by gottesmm** [comment]
@swift-ci Please smoke test os x platform

File: lib/TBDGen/TBDGen.cpp [hunk 1]
```diff
@@ -37,7 +37,8 @@ class TBDGenVisitor : public ASTVisitor<TBDGenVisitor> {
   StringSet &Symbols;
 
   void addSymbol(StringRef name) {
-    assert(Symbols.insert(name).second && "already inserted");
+    auto inserted = Symbols.insert(name).second;
+    assert(inserted && "already inserted");
```
Comment (slavapestov) [inline lib/TBDGen/TBDGen.cpp]: To avoid a warning with asserts off, add a (void) inserted; also

**Comment by huonw** [comment]
@swift-ci Please smoke test and merge.

**Comment by huonw** [comment]
@swift-ci Please smoke test and merge.

**Comment by huonw** [comment]
@swift-ci Please smoke test and merge.

**Comment by gottesmm** [comment]
@swift-ci Please smoke test and merge.

**Comment by gottesmm** [comment]
@swift-ci Please smoke test and merge.

**Pull request merged**